### ISSUE TITLE
[FEATURE REQ] Role Assignment - Allow adding of description

### ISSUE THREAD

**Issue by juergenbr** [Azure/azure-sdk-for-java#31421]
## Context
We want to add contextual information to role assignments using the description field. Since role assignments are not managed via the portal but via a centralised service we need a Java API to add a description when we create a new role assignment.

## Goal
Having the same fields available when creating a role assignment via the Java SDK as we have them in the Azure Portal. Specifically, we require the possibility to add a description to a role assignment when it gets created.

## Solution

add method "withDescription" to allow adding a description text.

```
azureResourceManager.accessManagement().roleAssignments()
                            .define(arm.resourceGroups().manager().internalContext().randomUuid())
                            .forObjectId(user.id)
                            .withRoleDefinition(roleDefinition.id())
                            .withSubscriptionScope(subscriptionId)
                            .withDescription("This is a description string")
                            .create();
```

**Comment by weidongxu-microsoft** [comment]
@juergenbr 
 
The "description" was added in api-version "2022-04-01". https://github.com/Azure/azure-rest-api-specs/blob/main/specification/authorization/resource-manager/Microsoft.Authorization/stable/2022-04-01/authorization-RoleAssignmentsCalls.json#L590-L593

Current SDK is at "2015-07-01" (the preview stable api-version).

We need to upgrade to 2022-04-01, before supporting the feature. It might take extra time.

**Comment by juergenbr** [comment]
@weidongxu-microsoft thanks for the update!
Any indication when this could become part of the SDK? We are currently in an implementation project where we require the description to be set.
Is it even recommended to rely on the Java SDK, considering how far behind some of the packages seem to be compared to the REST API?

**Comment by weidongxu-microsoft** [comment]
Next lib would be released about this week, on version 2.20.0

**Comment by weidongxu-microsoft** [comment]
2.20.0 released with the support
